Automattic/wp-calypso
# state-utils: Extract extendAction

#### Changes proposed in this Pull Request

* Extract `extendAction` into `state-utils`.

#### Testing instructions



This will test the `withoutNotice` function which uses `extendAction`.
* Open Pages in Calypso.
* Publish a new page, update and save an existing page, trash a page and restore a page from a previous version, and verify you don't see Calypso notices when you do those.

I couldn't find an example to test where we use `extendAction` over a `thunk` though 🤔 I think it should be fine as I didn't change any of the logic of the function, just the types.
